tools/topology/sof-tgl-sdw-max98373-rt5682.m4

```diff
@@ -70,6 +70,13 @@ define(`SMART_REF_CH_NUM', 2)
 define(`SMART_PCM_ID', 2)
 define(`SMART_PCM_NAME', `smart373-spk')
 
+# Maxim DSM UUID
+DECLARE_SOF_RT_UUID("Maxim DSM", maxim_dsm_comp_uuid, 0x0cd84e80, 0xebd3,
+                    0x11ea, 0xad, 0xc1, 0x02, 0x42, 0xac, 0x12, 0x00, 0x02);
+
+# comment out the following to fallback on smart_amp-test
+define(`SMART_UUID', maxim_dsm_comp_uuid)
```

Comment:
what does this mean? for smart_amp-test, you are using a different firmware anyways, aren't you?

Reply:
@plbossart but the UUIDs are different for the test dsm comp and maxim's comp. But I agree something seems missing with this change. Our internal tests dont use the maxim's DSM component. Commenting out will be a manual change that our tests cannot do.

See this issue for example: https://github.com/thesofproject/sof/issues/3603

Reply:
Just a note for switching between Maxim DSM and smart_amp-test, think we can enable both in the same firmware.

Reply:
@yongzhi1 yes we can but the maxim's library is not upstream isnt it?

Reply:
that was my point, we can't have in in-tree topology files that relies on out-of-tree firmware.

Reply:
Hi, @plbossart @ranj063 , I agree, since https://github.com/thesofproject/sof/pull/3374 was merged, thought this has been addressed. Sorry about that, let me know your suggestion, thanks!!

Reply:
@yongzhi1 this change will have to live on the product branch that contains the maxim's library. #3374 probably overlooked this.

Reply:
> @yongzhi1 this change will have to live on the product branch that contains the maxim's library. #3374 probably overlooked this.

Make sense, let me submit to the right branch or wait Maxim DSM to be merged.

Reply:
Hi, 
In,
https://github.com/thesofproject/sof/pull/3397
Maxim submitted their dummy algorithm library and last discussion on that PR says, dummy libdsm.a would be merged.

So, in that case: if tplg has this maxim UUID change along with dummy DSM.. there should be no problem in CI tests.

So, can you please consider this and comment if this tplg change can go in to master?.

(if product private branches has less and less delta from sof master, it will be better IMHO)

Reply:
do I get it right that a dummy and actual implementation have the same UUID? Is this intentional? that seems to defeat the purpose of a Unique ID :-(

Reply:
For product branch, we get same name of libdsm.a file with real algo - so that we can just replace  it. I thought, that way having same UUID will help.

Reply:
Should this be closed or is there more fixes/changes to come ?

Reply:
> Should this be closed or is there more fixes/changes to come ?

We prefer to get this patch merged with the understanding that the Maxim DSM module is stubbed in public domain. Either way @plbossart can make the call and close this PR.  There is no more fixes/changes, thanks!!